test/suites-ostree/healthy-upgrade-with-forced-rollback.robot

```diff
@@ -0,0 +1,71 @@
+*** Settings ***
```

Comment:
I think https://github.com/openshift/microshift/blob/main/test/suites-ostree/staged-failed-upgrade.robot tests this, too. How are the suites different? Do we need both?

Reply:
Staged-failed-upgrade is really similar. The difference is that staged-failed-upgrade forces the rollback without verifying the health of the upgraded os, which is half of what we want to ensure.

Reply:
We phrased some of the tickets for tests imprecisely, so I want to make sure that's a real requirement. I worry the scenario might turn out flakey if we try to require the initial boot to be healthy, but if there's a specific reason we need to require that for it to be a valid test we can just keep an eye on it.

cc @pmtk

Reply:
As I understood them, the test should deploy the y-1 release, upgrade to release-4.y, verify health, rollback, and again verify health. Is that correct?

Also, IIUC, an unhealthy state will block an upgrade. Is that accurate?

Reply:
That's not quite the scenario described in USHIFT-1108. The goal is to test that rollback works from y+1 to y. So we need y to be healthy, then upgrade to y+1, then trigger a rollback regardless of how the upgrade goes, then verify y is healthy again.

That other RF suite file does all of that.

Reply:
I don't think `staged-failed-upgrade.robot` rolls back: it fails on first boot of new deployment, but on second it becomes healthy.

It's `failed-upgrade.robot` that consistently fails, causing system to roll back.
However, currently the test does that by preventing backup.
In https://github.com/openshift/microshift/pull/2184 I change that to have two separate ([1](https://github.com/openshift/microshift/pull/2184/files#diff-f7bef174572b9938e8a98a36522c46782a66013bd4ce4c77b2363f536731225f), [2](https://github.com/openshift/microshift/pull/2184/files#diff-23f9418f94001d3da07283921bc1cdb8a252f2a99b4ac6b18f9a54799186bbb1)) scenarios based on [single RF file with a reason variable](https://github.com/openshift/microshift/pull/2184/files#diff-e6691d89f9a5bc89929fe7c6f5561b29f52c2aae793a7d21598e0420715b3059)